### PR TITLE
internal: a number of code simplifications

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

File: crates/ide-diagnostics/src/handlers/json_is_not_rust.rs [hunk 1]
```diff
@@ -99,76 +99,66 @@ pub(crate) fn json_in_items(
...
-                                        config.insert_use.prefix_kind,
-                                        config.prefer_no_std,
-                                    ) {
-                                        insert_use(
-                                            &scope,
-                                            mod_path_to_ast(&it),
-                                            &config.insert_use,
-                                        );
-                                    }
+            if let Ok(serde_json::Value::Object(it)) = serde_json::from_str(&node_string) {
```
Comment (lnicola) [inline crates/ide-diagnostics/src/handlers/json_is_not_rust.rs:102]: :+1:

File: crates/project-model/src/workspace.rs [hunk 2]
```diff
@@ -366,11 +366,11 @@ impl ProjectWorkspace {
                 _ => None,
             })
             .collect();
-        let ref mut outputs = match WorkspaceBuildScripts::run_once(config, &cargo_ws, progress) {
+        let outputs = &mut (match WorkspaceBuildScripts::run_once(config, &cargo_ws, progress) {
```
Comment (Veykril) [inline crates/project-model/src/workspace.rs:369]: Are the extra parentheses even necessary here?

Comment (danieleades) [reply crates/project-model/src/workspace.rs:369]: negative. will update

**Pull request merged**